calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java

```diff
@@ -239,7 +243,7 @@ public List filterEvents(List events, String[] customFieldNames, String tzid) th
 			// Raw + calendar/owner TZ's offset
 			ZonedDateTime srcZonedDateTime = startInstant.atZone(srcZoneId);
 			long millis = startInstant.plusMillis(srcZonedDateTime.getOffset().getTotalSeconds() * 1000).toEpochMilli();
-			TimeZone tz = TimeZone.getTimeZone(tzid);
+			TimeZone tz = TimeZone.getTimeZone(srcZoneId);
```

Comment:
Why srcZoneId here? What's the differerence?

Reply:
I found this related bug while testing, on the top of that method there's this:

`		if (tzid != null) {
			srcZoneId = ZoneId.of(tzid);
		} else {
			srcZoneId = dstZoneId;
		}`

So tzid will be null on this case and it will throw a NPE.